docs/articles/documentation/guides/continuous-integration/github-actions-and-browserstack.md

```diff
@@ -0,0 +1,249 @@
+---
+layout: docs
+title: Run Tests on BrowserStack with GitHub Actions
+permalink: /documentation/guides/continuous-integration/github-actions-and-browserstack.html
+---
+# Run Tests on BrowserStack with GitHub Actions
+
+This topic describes how to use the [Run TestCafe action](https://github.com/DevExpress/testcafe-action) to integrate TestCafe into the [GitHub Actions](https://docs.github.com/en/actions/automating-your-workflow-with-github-actions) build process. Tests are executed in the [BrowserStack](https://automate.browserstack.com/) cloud testing service.
+
+* [Step 1 - Create a Workflow](#step-1---create-a-workflow)
+* [Step 2 - Create a Job](#step-2---create-a-job)
+* [Step 3 - Provide Your BrowserStack Credentials](#step-3---provide-browserstack-credentials)
+* [Step 4 - Add a Step that Fetches the Repository](#step-4---add-a-step-that-fetches-the-repository)
+* [Step 5 - Add a Step to Install the TestCafe BrowserStack Plugin](#step-5---add-a-step-to-install-testcafe-browserstack-plugin)
+* [Step 6 - Add a Step that Runs TestCafe](#step-6---add-a-step-that-runs-testcafe)
+* [Action Options](#action-options)
+  * [args](#args)
+  * [version](#version)
+* [Example](#example)
+
+## Step 1 - Create a Workflow
+
+Create a YAML file (for instance, `testcafe-workflow.yml`) in the `.github/workflows` directory in your repository.
+
+Specify the [workflow name](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#name) and the [event](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#on) that triggers this workflow.
+
+```yml
+name: End-to-End Tests
+on: [push]
+```
+
+In this example, the workflow runs when you push changes to the repository.
+
+## Step 2 - Create a Job
+
+Create a job that runs the TestCafe tests.
+
+Provide the [job name](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idname) and specify the [type of machine](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idruns-on) that should run the job.
+
+```yml
+name: End-to-End Tests
+on: [push]
+
+jobs:
+  test:
+    name: Run TestCafe Tests
+    runs-on: windows-latest
```

Comment:
It's quite a nitpick, but when we use cloud browsers, we can use any base OS with no or little differences.

Reply:
@AndreyBelym So it seems. I tried launching a simple test for a matrix `os: [ubuntu-latest, windows-latest, macos-latest]` and all of them ran fine. MacOS has a [lower limit for concurrent jobs](https://docs.github.com/en/actions/getting-started-with-github-actions/about-github-actions#usage-limits) though.

I guess we could mention that all OS's are pretty much interchangable in this case. Should I add a config example for a [self-hosted runner](https://docs.github.com/en/actions/hosting-your-own-runners/about-self-hosted-runners) as well?

Reply:
Well, I think instructions for connecting self-hosted runners can be useful for some people. Actually, we use self-hosted runners to run our functional tests with BrowserStack due to some networking problems.

Reply:
Added [here](https://github.com/DevExpress/testcafe/pull/5344/commits/523a569bddd4a49ecd2051d3ae0ca476ad3b8f19) 

Please review